adcc/LazyMp.py

```diff
@@ -60,39 +57,80 @@ def __getattr__(self, attr):
 
     def df(self, space):
         """Delta Fock matrix"""
-        # TODO This is so cheap to compute that it does not really deserve
-        #      any caching
-        return self.__cpp.df(space)
+        hf = self.reference_state
+        s1 = space[:2]
+        s2 = space[2:]
```

Comment:
I would write a function that (a) checks that the spaces are sensible and (b) splits them up into chunks. What I have in mind is:
```
s1, s2 = split_spaces(space, 2)
```
That you can then reuse for all other similar tasks in the other functions as well.

Reply:
Implemented now, but needs some "cleanup"/improvement 😄

Reply:
Resolved.